steamclog/src/main/java/com/steamclock/steamclog/Steamclog.kt

```diff
@@ -48,13 +51,41 @@ object SteamcLog {
         // Don't plant yet; fileWritePath required before we can start writing to ExternalLogFileDestination
     }
 
-    fun initWith(config: Config) {
+    fun initWith(context: Context, config: Config) {
         this.config = config
-        this.config.fileWritePath?.let {
+
+        // Setup ExternalLogFileDestination
+        if (this.config.fileWritePath != null && this.config.fileWritePath!!.exists()) {
             updateTree(externalLogFileTree, true)
-        } ?: run {
-            logInternal(LogLevel.Warn, "fileWritePath given was null; cannot log to external file")
+        } else {
+            // We have seen issues where some devices are failing to support some of the default file
+            // paths (ie. externalCacheDir); the code below attempts to catch that case and report it
+            // proactively to Sentry as an error once per app install.
+            val sentryErrorTitle = "Steamclog could not create external logs"
+            logInternal(LogLevel.Info, "File path ${this.config.fileWritePath} is invalid")
+
+            try {
+                // Attempt to log error only once to avoid overwhelming Sentry.
+                // runBlocking usage was recommended in the google docs as the way to synchronously
+                // call the datastore methods; we need to use this with caution, but
+                // https://developer.android.com/topic/libraries/architecture/datastore#synchronous
+                val dataStore = SClogDataStore(context)
+                val alreadyReportedFailure = runBlocking { dataStore.getHasReportedFilepathError.first() }
```

Comment:
Good call. I didn't make the entire method suspend (Timber documentation for planting/updating trees has historically been called from the main thread I believe, at least it's been that way since first implementation)... but I did move the code that checked for file access errors and called `runBlocking` on the dataStore to be in a new coroutine scope.